### ISSUE TITLE
aws_opsworks_rds_db_instance needs some time to become available

### ISSUE THREAD

**Issue by hashibot** [terraform-providers/terraform-provider-aws#615]
_This issue was originally opened by @steveh as hashicorp/terraform#12786. It was migrated here as part of the [provider split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/). The original body of the issue is below._

<hr>

An instance of `aws_opsworks_rds_db_instance` (not the RDS database itself; the Opsworks reference to it within an Opsworks stack) is not immediately available when created, it takes 30 seconds or so.

I've noticed other resources can retry for a little while until dependent resources become available; I think this needs to happen for `aws_opsworks_application.data_source_arn`.

### Terraform Version
Terraform v0.8.8

### Affected Resource(s)
- aws_opsworks_rds_db_instance
- aws_opsworks_application

### Terraform Configuration Files
```hcl
resource "aws_db_instance" "main" {
}

resource "aws_opsworks_rds_db_instance" "main" {
  rds_db_instance_arn = "${aws_db_instance.main.arn}"
}

resource "aws_opsworks_application" "main" {
  data_source_arn = "${aws_db_instance.main.arn}"
}
```

### Expected Behavior
`aws_opsworks_application` created successfully

### Actual Behavior
```
* aws_opsworks_application.main: ResourceNotFoundException: unable to find rds instance with ARN arn:aws:rds:ap-southeast-2:XXX:db:XXX
	status code: 400, request id: XXX
```

Repeat attempts are successful.

### Steps to Reproduce
1. Create an RDS instance
2. Create an Opsworks RDS resource and assign it to an application simultaneously
3. Retry

**Comment by bflad** [comment]
Hi @steveh 👋 Thank you for reporting this awhile ago and hopefully you figured out a solution in the meantime.

In this case, to have Terraform properly order operations, the configuration would need to be updated to the following:

```hcl
resource "aws_db_instance" "main" {
}

resource "aws_opsworks_rds_db_instance" "main" {
  rds_db_instance_arn = "${aws_db_instance.main.arn}"
}

resource "aws_opsworks_application" "main" {
  data_source_arn = "${aws_opsworks_rds_db_instance.main.rds_db_instance_arn}"
}
```

Which would ensure the `aws_opsworks_rds_db_instance` resource is created before attempting the `aws_opsworks_application` resource.

If you are still having trouble on recent versions of Terraform CLI and the Terraform AWS Provider, please feel free to create a new bug report following the issue template and we can take a fresh look. 👍

**Comment by Velgus** [comment]
This bug is still not resolved as of Terraform 0.12.21.

The issue isn't to do with the order of resources, but the fact that `aws_opsworks_rds_db_instance` gets assigned instantly, but does not actually become available immediately to other resources.

As a result, you can end up with output such as:
```
module.opsworks_stack.aws_opsworks_rds_db_instance.stack_rds: Creation complete after 0s [id=arn:aws:rds:us-west-2:<db_number>:db:stage1-new-rds-cluster-<db_identifier>]

Error: Error creating OpsWorks application: ResourceNotFoundException: unable to find rds instance with ARN arn:aws:rds:us-west-2:<db_number>:db:stage1-new-rds-cluster-1

  on ../../modules/opsworks_app/apps.tf line 1, in resource "aws_opsworks_application" "app":
   1: resource "aws_opsworks_application" "app" {
```

If you then run a Terraform (or in my case, Atlantis) plan and apply again a few moments later, it will complete successfully.
